src/strategies/erc721-with-tokenid-range-weights-simple/index.ts

```diff
@@ -19,48 +39,66 @@ export async function strategy(
   snapshot
 ) {
   const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+  const { tokenIdWeightRanges, defaultWeight } = options;
+  let erc721WeightedBalance = {};
+
+  const getDataFromBlockChain = async (
+    contractCalls: [string, string, [number]][]
+  ) => multicall(network, provider, abi, contractCalls, { blockTag });
+
+  const filterUnusedAddresses = (addressesToFilter: string[]): string[] =>
+    addressesToFilter.filter((address: string) =>
+      addresses
+        .map((address: string) => address.toLowerCase())
+        .includes(address.toLowerCase())
+    );
+
+  const multiplyOccurrencesByWeights = async (
+    contractCallResponse: [string, { owner: string }],
+    weights: number[]
+  ) =>
+    contractCallResponse
+      .map((address, index) => Array(weights[index]).fill(address[0]))
+      .flat();
+
+  const countOccurrences = (array: string[]) =>
+    array.reduce(
+      (prev, curr) => (prev[curr] ? ++prev[curr] : (prev[curr] = 1), prev),
+      {}
+    );
+
+  const getCustomRangeBalance = async (): Promise<{
+    [address: string]: number;
+  }> => {
+    const { ids, weights } = flattenTokenIdWeightRanges(tokenIdWeightRanges);
+    const contractCalls = ids.map((id: number) => [
+      options.address,
+      'ownerOf',
+      [id]
+    ]) as [string, string, [number]][];
+    const customRangeResponse = await getDataFromBlockChain(contractCalls);
+    const customRangeResponseWeighted = await multiplyOccurrencesByWeights(
+      customRangeResponse,
+      weights
+    );
+    const customRangeResponseWeightedFiltered = filterUnusedAddresses(
+      customRangeResponseWeighted
+    );
+
+    return countOccurrences(customRangeResponseWeightedFiltered);
+  };
+
+  if (defaultWeight)
+    erc721WeightedBalance = await erc721WithMultiplier(
+      space,
+      network,
+      provider,
+      addresses,
+      { ...options, multiplier: defaultWeight },
+      snapshot
+    );
```

Comment:
If defaultWeight exist, you are adding erc721 balance to range? better to use erc721WithMultiplier in your space seperately, result of voting power will be sum of these two strategies